app/src/main/java/com/sun_asterisk/moviedb_50/data/source/remote/fetchjson/ParseJsonToMovie.kt

```diff
@@ -0,0 +1,67 @@
+package com.sun_asterisk.moviedb_50.data.source.remote.fetchjson
+
+import com.sun_asterisk.moviedb_50.data.model.Genres
+import com.sun_asterisk.moviedb_50.data.model.Movie
+import org.json.JSONArray
+import org.json.JSONException
+import org.json.JSONObject
+
+
+class ParseJsonToMovie {
+
+    fun parseJsonToData(jsonObject: JSONObject): MutableList<Movie>? {
+        val movieList: MutableList<Movie> = mutableListOf()
+        try {
+            val jsonArray = jsonObject.getJSONArray(Movie.MovieEntry.MOVIE)
+            for (i in 0 until jsonArray.length()) {
+                val movieJson = jsonArray.getJSONObject(i)
+                val movie: Movie? = movieJson?.let { parseJsonToObject(it) }
+                movie?.let { movieList.add(it) }
+            }
+        } catch (e: JSONException) {
+            e.printStackTrace()
+        }
+        return movieList
+    }
+
+    private fun parseJsonToObject(movieJson: JSONObject): Movie? {
+        try {
+            val id = movieJson.optInt(Movie.MovieEntry.ID)
+            val title = movieJson.optString(Movie.MovieEntry.TITLE)
+            val overview = movieJson.optString(Movie.MovieEntry.OVERVIEW)
+            val posterPath = movieJson.optString(Movie.MovieEntry.POSTER_PATH)
+            val backdropPath = movieJson.optString(Movie.MovieEntry.BACKDROP_PATH)
+            val voteCount = movieJson.optInt(Movie.MovieEntry.VOTE_COUNT)
+            val voteAverage = movieJson.optDouble(Movie.MovieEntry.VOTE_AVERAGE)
+            val popularity = movieJson.optDouble(Movie.MovieEntry.POPULARITY)
+            val releaseDate = movieJson.optString(Movie.MovieEntry.RELEASE_DATE)
+            val genre = getGenresIdArray(movieJson.optJSONArray(Genres.GenresEntry.GENRES))
+
+            return Movie(
+                id,
+                title,
+                overview,
+                posterPath,
+                backdropPath,
+                voteCount,
+                voteAverage,
+                popularity,
+                releaseDate,
+                genre
+            )
+        } catch (e: Exception) {
+            e.printStackTrace()
+        }
+        return null
+    }
+
+    private fun getGenresIdArray(strArr: JSONArray?): IntArray? {
+        val numbers = strArr?.let { IntArray(it.length()) }
+        strArr?.let {
+            for (i in 0 until it.length()) {
+                numbers?.set(i, it.optInt(i))
+            }
+        }
+        return numbers
+    }
```

Comment:
check null 1 lần thay vì nhiều lần 
```suggestion
    private fun getGenresIdArray(strArr: JSONArray?): IntArray? {
        strArr?:return null 
        val numbers = strArr.let { IntArray(it.length()) }
        strArr.let {
            for (i in 0 until it.length()) {
                numbers?.set(i, it.optInt(i))
            }
        }
        return numbers
    }
```